nio/src/main/scala/zio/nio/file/Files.scala

```diff
@@ -279,9 +279,15 @@ object Files {
 
   def list(path: Path): ZStream[Blocking, Exception, Path] =
     ZStream
-      .fromEffect(effectBlocking(JFiles.list(path.javaPath).iterator()).refineToOrDie[IOException])
-      .flatMap(fromJavaIterator)
+      .fromJavaIteratorManaged(
+        ZManaged
+          .make(effectBlocking(JFiles.list(path.javaPath)))(stream => ZIO.effectTotal(stream.close()))
+          .map(_.iterator())
+      )
       .map(Path.fromJava)
+      .refineOrDie {
+        case io: IOException => io
+      }
```

Comment:
```suggestion
      .refineToOrDie[IOException]
```

Reply:
that was my first reaction but this method is only available on `ZIO` but not on `ZStream`